### ISSUE TITLE
Fail to use magic to generate report

### ISSUE THREAD

**Issue by HenryLeongStat** [vatlab/sos-notebook#122]
<img width="1182" alt="screen shot 2018-09-05 at 1 25 22 pm" src="https://user-images.githubusercontent.com/26916551/45113153-2b119b00-b10f-11e8-9ea9-c59ab34b5257.png">

**Comment by HenryLeongStat** [comment]
But `$ sos convert xxx.ipynb yyy.html --template sos-report` works fine.

**Comment by BoPeng** [comment]
Yes, this is broken due to the addition of parameter `--execute` to `sos convert`. The problem here is that we cannot test magic `%sossave` because the test suite does not have the .ipynb file to convert.

**Comment by HenryLeongStat** [comment]
Seems it still fails...
<img width="1185" alt="screen shot 2018-09-05 at 2 26 26 pm" src="https://user-images.githubusercontent.com/26916551/45116234-b131df80-b117-11e8-8d81-c78871ac2d36.png">

**Comment by HenryLeongStat** [comment]
Pulse I got ton of error msg when I started jupyter-notebook:
```
...
[1386:775:0905/142423.188658:ERROR:in_progress_cache_impl.cc(191)] Cache is not initialized, cannot RetrieveEntry.
[1386:775:0905/142423.188685:ERROR:in_progress_cache_impl.cc(191)] Cache is not initialized, cannot RetrieveEntry.
[1386:775:0905/142423.188718:ERROR:in_progress_cache_impl.cc(191)] Cache is not initialized, cannot RetrieveEntry.
[1386:775:0905/142423.188745:ERROR:in_progress_cache_impl.cc(191)] Cache is not initialized, cannot RetrieveEntry.
[1386:775:0905/142423.188778:ERROR:in_progress_cache_impl.cc(191)] Cache is not initialized, cannot RetrieveEntry.
[1386:775:0905/142423.188809:ERROR:in_progress_cache_impl.cc(191)] Cache is not initialized, cannot RetrieveEntry.
[1386:775:0905/142423.188837:ERROR:in_progress_cache_impl.cc(191)] Cache is not initialized, cannot RetrieveEntry.
[1386:775:0905/142423.188870:ERROR:in_progress_cache_impl.cc(191)] Cache is not initialized, cannot RetrieveEntry.
[1386:775:0905/142423.188897:ERROR:in_progress_cache_impl.cc(191)] Cache is not initialized, cannot RetrieveEntry.
[1386:775:0905/142423.188930:ERROR:in_progress_cache_impl.cc(191)] Cache is not initialized, cannot RetrieveEntry.
[1386:775:0905/142423.188962:ERROR:in_progress_cache_impl.cc(191)] Cache is not initialized, cannot RetrieveEntry.
[1386:775:0905/142423.188991:ERROR:in_progress_cache_impl.cc(191)] Cache is not initialized, cannot RetrieveEntry.
[1386:775:0905/142423.189024:ERROR:in_progress_cache_impl.cc(191)] Cache is not initialized, cannot RetrieveEntry.
[1386:775:0905/142423.189052:ERROR:in_progress_cache_impl.cc(191)] Cache is not initialized, cannot RetrieveEntry.
[1386:775:0905/142423.189084:ERROR:in_progress_cache_impl.cc(191)] Cache is not initialized, cannot RetrieveEntry.
...
```

**Comment by BoPeng** [comment]
Sorry, failed to push this one. Try again

**Comment by BoPeng** [comment]
Your warning was from another program, which is unrelated to jupyter or from some jupyter component.

**Comment by HenryLeongStat** [comment]
It doesn't show any error msg after the new push. 
<img width="289" alt="screen shot 2018-09-05 at 2 35 13 pm" src="https://user-images.githubusercontent.com/26916551/45116623-ed197480-b118-11e8-80cc-11711a19c9ba.png">

However, the output `.html` file is empty.
<img width="1174" alt="screen shot 2018-09-05 at 2 35 34 pm" src="https://user-images.githubusercontent.com/26916551/45116647-f73b7300-b118-11e8-9e2d-022c99b7b66d.png">

For using command line, the output is correct:
<img width="400" alt="screen shot 2018-09-05 at 2 36 21 pm" src="https://user-images.githubusercontent.com/26916551/45116667-133f1480-b119-11e8-88b0-d70c0e20903d.png">

**Comment by BoPeng** [comment]
That is what I have talked about today, namely the default template hide all code cells. Move your cursor to the top left corner to display it (we can make it more obvious) or use tags to show cells.

**Comment by HenryLeongStat** [comment]
But those cells have tags already. By using command line `$ sos convert xxx.ipynb yyy.html --template sos-report`, it works fine. 

<img width="863" alt="screen shot 2018-09-05 at 3 05 38 pm" src="https://user-images.githubusercontent.com/26916551/45118136-2c49c480-b11d-11e8-892c-ac8a8020f305.png">

**Comment by BoPeng** [comment]
Ohmm, please PM me the notebook.

**Comment by HenryLeongStat** [comment]
Sent via gitter

**Comment by BoPeng** [comment]
Works here...

**Comment by HenryLeongStat** [comment]
Weird... Got another error msg this time...
<img width="1181" alt="screen shot 2018-09-05 at 3 29 03 pm" src="https://user-images.githubusercontent.com/26916551/45119243-7a13fc00-b120-11e8-957a-309617b7b50a.png">

But on terminal:
```
[NbConvertApp] Converting notebook /Users/manchongleong/Untitled.ipynb to html
[NbConvertApp] Writing 309290 bytes to /Users/manchongleong/Untitled.html
```

and `Untitled.html` is blank like:
<img width="1218" alt="screen shot 2018-09-05 at 3 30 28 pm" src="https://user-images.githubusercontent.com/26916551/45119291-a3348c80-b120-11e8-8f1c-06a7c58fde91.png">

**Comment by BoPeng** [comment]
You should at least be able to open the generated HTML file from a text editor, right?

**Comment by HenryLeongStat** [comment]
Yes, and it is not empty...

**Comment by BoPeng** [comment]
Then send me the html file?

**Comment by BoPeng** [comment]
Also, please let me know the steps to reproduce that red warning. I have seen it from time to time but do not know how to trigger it. It looks like a version mismatch to me.

**Comment by HenryLeongStat** [comment]
Sent via gitter.
The red warning is produced by using `$ jupyter notebook Untitled14.ipynb`.
Then execute `%sossave --to html --force`

**Comment by BoPeng** [comment]
Works here, so it looks like a version mismatch problem on your side.
